### ISSUE TITLE
[Error] Intelephense server crashed 5 times in the last 3 minutes

### ISSUE THREAD

**Issue by jakebathman** [bmewburn/intelephense#66]
**OS**: Windows 10.0.16299
**VS Code**: 1.17.2
**Intelephense**: 0.8.0

I've been receiving this notice from VS Code upon opening a project today. I enabled verbose log output and here's what it shows (some paths redacted):

```
[Trace - 1:44:55 PM] Sending request 'initialize - (0)'.
Params: {
    "processId": 5756,
    "rootPath": "c:\\Users\\jbathman\\Documents\\path\\to\\folder",
    "rootUri": "file:///c%3A/Users/jbathman/Documents/path/to/folder",
    "capabilities": {
        "workspace": {
            "didChangeConfiguration": {
                "dynamicRegistration": true
            },
            "didChangeWatchedFiles": {
                "dynamicRegistration": true
            },
            "symbol": {
                "dynamicRegistration": true
            },
            "executeCommand": {
                "dynamicRegistration": true
            }
        },
        "textDocument": {
            "synchronization": {
                "dynamicRegistration": true,
                "willSave": true,
                "willSaveWaitUntil": true,
                "didSave": true
            },
            "completion": {
                "dynamicRegistration": true,
                "completionItem": {
                    "snippetSupport": true
                }
            },
            "hover": {
                "dynamicRegistration": true
            },
            "signatureHelp": {
                "dynamicRegistration": true
            },
            "definition": {
                "dynamicRegistration": true
            },
            "references": {
                "dynamicRegistration": true
            },
            "documentHighlight": {
                "dynamicRegistration": true
            },
            "documentSymbol": {
                "dynamicRegistration": true
            },
            "codeAction": {
                "dynamicRegistration": true
            },
            "codeLens": {
                "dynamicRegistration": true
            },
            "formatting": {
                "dynamicRegistration": true
            },
            "rangeFormatting": {
                "dynamicRegistration": true
            },
            "onTypeFormatting": {
                "dynamicRegistration": true
            },
            "rename": {
                "dynamicRegistration": true
            },
            "documentLink": {
                "dynamicRegistration": true
            }
        }
    },
    "initializationOptions": {
        "storagePath": "C:\\Users\\jbathman\\AppData\\Roaming\\Code\\User\\workspaceStorage\\76d8f907b525a2d1253d63ef5a3bd9ee\\bmewburn.vscode-intelephense-client"
    },
    "trace": "verbose"
}


[Trace - 1:44:56 PM] Received notification 'window/logMessage'.
Params: {
    "type": 3,
    "message": "Initialising"
}


[Info  - 1:44:56 PM] Initialising
undefined:1



SyntaxError: Unexpected end of JSON input
    at JSON.parse (<anonymous>)
    at fs.readFile (C:\Users\jbathman\.vscode\extensions\bmewburn.vscode-intelephense-client-0.8.0\server\node_modules\intelephense\lib\cache.js:68:26)
    at C:\Users\jbathman\.vscode\extensions\bmewburn.vscode-intelephense-client-0.8.0\server\node_modules\graceful-fs\graceful-fs.js:78:16
    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:439:3)
[Info  - 1:45:01 PM] Connection to server got closed. Server will restart.
```
This repeats a few times before ultimately failing with this message instead of the "server will restart" one: 
```
[Error - 1:45:17 PM] Connection to server got closed. Server will not be restarted.
``

Not exactly sure what's going on, so I'll see if anyone else has this issue or if the developer has any ideas.

**Comment by bmewburn** [comment]
Looks like the file cache crashed. Do you have a lot of files in the project?

**Comment by jakebathman** [comment]
Yep, there's a ton. I can get an exact number and type breakdown if it'd help?

**Comment by bmewburn** [comment]
A count would be useful. As a workaround you could see if you could exclude some of it using  `files.exclude` and delete the contents of `C:\\Users\\jbathman\\AppData\\Roaming\\Code\\User\\workspaceStorage\\76d8f907b525a2d1253d63ef5a3bd9ee\\bmewburn.vscode-intelephense-client` to force a re-index.

**Comment by jakebathman** [comment]
Here's the `cloc` output for the directory, I'll give exclude a try to make sure it doesn't choke.

```
   28192 text files.
   23886 unique files.
   11076 files ignored.

http://cloc.sourceforge.net v 1.60  T=713.63 s (24.1 files/s, 4310.9 lines/s)
--------------------------------------------------------------------------------
Language                      files          blank        comment           code
--------------------------------------------------------------------------------
PHP                            9617         151851         305051        1241675
Javascript                     4102         118856         197262         795684
CSS                             670           6037           4023         103426
HTML                            400          10471           1916          86154
LESS                            105           2735           3431          13412
XML                            2078            144             33           7501
XSD                              16            213             48           6106
YAML                            133            480            188           5566
CoffeeScript                     26            893            551           3451
SASS                              9            499            271           1692
SQL                               5             74             28            979
C                                 2            199             40            966
Ant                              14            160              3            952
make                             14            214            141            942
DOS Batch                         2             46              2            332
Bourne Shell                     20             89             70            310
Python                            5            212            330            262
C/C++ Header                      3             48             54            160
m4                                2             22             22             82
Ruby                              2              2              0             20
Bourne Again Shell                1              1              1             10
--------------------------------------------------------------------------------
SUM:                          17226         293246         513465        2269682
--------------------------------------------------------------------------------
```

**Comment by DanStevens** [comment]
I'm getting this error as well. Not sure how to get a count of files in a directory by type as above.

**Comment by DanStevens** [comment]
I just cleared the workspaceStorage directory (there was over 500MB of data). The intelephense indexer appears to have completed for my workspace now. 

[Info  - 17:31:10] Indexing ended | 12008 files | 114.315 s

**Comment by arturock** [comment]
A basic ignore config to remove tests:
```js
    "files.exclude": {
        ...
        "**/vendor/**/Tests" : true,
        "**/vendor/**/tests" : true,
        "**/vendor/**/test"  : true
    },
```
but you may want to ignore some dependencies completely depending on your project.
Also, there's the command `Intelephense: Clear Cache and Reload` (`intelephense.clear.cache`) for the caching, that was what I used when I faced this same error and it presumably got fixed.

**Comment by ksidibe** [comment]
I'm having the same problem and excluding files like suggested in earlier in the thread did not solve the issue.
OS: MacOs 10.12.6
VS Code: 1.18.0
intelephense: 0.8.0
Items Count In Project: 32,760
Project Size on Disk: 298.2 MB
Any other suggestions?
Thanks.

**Comment by bmewburn** [comment]
Should now be fixed in v0.8.1 . JSONStream is now used for writing large arrays to disk and there's improved error handling.

**Comment by probfacile** [comment]
I am on version 1.01 and the problem is still there
